nixos/modules/services/security/oauth2_proxy_nginx.nix

```
@@ -13,6 +13,19 @@ in
         The address of the reverse proxy endpoint for oauth2_proxy
       '';
     };
+
+    domain = mkOption {
+      type = types.str;
+      default = lib.head cfg.virtualHosts;
+      defaultText = literalExpression "lib.head config.services.oauth2_proxy.nginx.virtualHosts";
+      description = lib.mdDoc ''
+        The domain under which the oauth2_proxy will be accesible and the path of cookies are set to.
+        If oauth2-proxy is configured with `cookie.domain` or multiple virtualHosts that are not on the same domain,
+        this setting must be set to ensure back-redirects are working properly.
```

```suggestion
        This setting must be set to ensure back-redirects are working properly 
        if oauth2-proxy is configured with `cookie.domain` or multiple `virtualHosts`
        that are not on the same domain.
```

Not sure if better, re-flow the text as you see fit.

---

I think this description will benefit from `{option}` notation.